### PR TITLE
ngx-build: added new patched needed for balancer_by_lua_* to function

### PR DESCRIPTION
**Pull request by dndx**
optimally.

### CODE REVIEW

File: ngx-build [hunk 1]
```diff
@@ -495,6 +495,24 @@ sub apply_patches {
...
+        chdir "nginx-$version" or die "cannot switch to nginx-$version\n";
+        shell("patch -p1 < $root/../openresty/patches/nginx-$version-stream_proxy_get_next_upstream_tries.patch");
+        chdir ".." or die "cannot switch to ..\n";
+    }
+
+    if ($ver ge '001013002') {
+        chdir "nginx-$version" or die "cannot switch to nginx-$version\n";
+        shell("patch -p1 < $root/../openresty/patches/nginx-$version-stream_proxy_timeout_fields.patch");
+        chdir ".." or die "cannot switch to ..\n";
+    }
```
Comment (agentzh) [inline ngx-build]: Seems like we can merge these patches under a single `if` condition?

File: ngx-build [hunk 2]
```diff
@@ -495,6 +495,24 @@ sub apply_patches {
         shell("patch -p1 < $root/../openresty/patches/nginx-$version-balancer_status_code.patch");
         chdir ".." or die "cannot switch to ..\n";
     }
+
+    if ($ver ge '001013002') {
```
Comment (agentzh) [inline ngx-build]: Maybe it should be `001013003` here instead?

Comment (dndx) [reply ngx-build]: Ah, I didn't knew `ge` means "greater than or equal" before, thought it was strictly greater than. Fixed.

**Comment by agentzh** [comment]
Merged. Thanks!

**Pull request closed without merging**